server/plugin.go

```diff
@@ -23,29 +24,30 @@ type Plugin struct {
 	// setConfiguration for usage.
 	configuration *configuration
 
-	badWords map[string]bool
-}
-
-func (p *Plugin) WordIsBad(word string) bool {
-	_, ok := p.badWords[strings.ToLower(removeAccents(word))]
-	return ok
+	badWordsRegex *regexp.Regexp
 }
 
 func (p *Plugin) FilterPost(post *model.Post) (*model.Post, string) {
-	configuration := p.getConfiguration()
-
-	message := post.Message
-	words := strings.Split(message, " ")
-	for i, word := range words {
-		if p.WordIsBad(word) {
-			if configuration.RejectPosts {
-				return nil, "Profane word not allowed: " + word
-			}
-			words[i] = strings.Repeat(configuration.CensorCharacter, len(word))
+	postWithoutAccents := removeAccents(post.Message)
+
+	if p.badWordsRegex.MatchString(postWithoutAccents) {
```

Comment:
nit: 1/5 `if !match { return post, nil }` - for better typography

Reply:
Agree